app/src/lib/stores/git-store.ts

```diff
@@ -320,6 +324,20 @@ export class GitStore extends BaseStore {
     return allBranchesWithUpstream
   }
 
+  private async checkPullWithRebase() {
+    const result = await getConfigValue(this.repository, 'pull.rebase')
+
+    if (result === null || result === '') {
+      this.pullWithRebase = undefined
+    } else if (result === 'true') {
+      this.pullWithRebase = true
+    } else if (result === 'false') {
+      this.pullWithRebase = false
+    } else {
+      log.warn(`Unexpected value found for pull.rebase in config: '${result}'`)
```

Comment:
If we get to the `else` clause, should we be explicitly setting `this.pullWithRebase` to `undefined`, clearing what ever the previous value is?

Reply:
Good shout, fixed in  ff462628c